diff --git a/CMake/FindBrotli.cmake b/CMake/FindBrotli.cmake
index bd3363cce0..8cf2914bad 100644
--- a/CMake/FindBrotli.cmake
+++ b/CMake/FindBrotli.cmake
@@ -42,7 +42,7 @@ if(CURL_USE_PKGCONFIG AND
    NOT DEFINED BROTLI_INCLUDE_DIR AND
    NOT DEFINED BROTLICOMMON_LIBRARY AND
    NOT DEFINED BROTLIDEC_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_brotli ${_brotli_pc_requires})
 endif()
 
@@ -85,7 +85,7 @@ endif()
 
 if(BROTLI_FOUND)
   if(NOT TARGET CURL::brotli)
-    add_library(CURL::brotli INTERFACE IMPORTED)
+    add_library(CURL::brotli INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::brotli PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_brotli_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_brotli_CFLAGS}"
diff --git a/CMake/FindCares.cmake b/CMake/FindCares.cmake
index 26efdf87ff..f424ad4e90 100644
--- a/CMake/FindCares.cmake
+++ b/CMake/FindCares.cmake
@@ -40,11 +40,11 @@ set(_cares_pc_requires "libcares")
 if(NOT DEFINED CARES_INCLUDE_DIR AND
    NOT DEFINED CARES_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_cares ${_cares_pc_requires})
   endif()
   if(NOT _cares_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(c-ares CONFIG QUIET)
+    find_package(c-ares CONFIG QUIET GLOBAL)
   endif()
 endif()
 
@@ -121,7 +121,7 @@ if(CARES_FOUND)
   endif()
 
   if(NOT TARGET CURL::cares)
-    add_library(CURL::cares INTERFACE IMPORTED)
+    add_library(CURL::cares INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::cares PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_cares_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_cares_CFLAGS}"
diff --git a/CMake/FindGSS.cmake b/CMake/FindGSS.cmake
index 9237fb30b1..8c9776e86f 100644
--- a/CMake/FindGSS.cmake
+++ b/CMake/FindGSS.cmake
@@ -50,7 +50,7 @@ set(_gss_LIBRARY_DIRS "")
 # Try to find library using system pkg-config if user did not specify root dir
 if(NOT GSS_ROOT_DIR AND NOT "$ENV{GSS_ROOT_DIR}")
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_search_module(_gss ${_gnu_modname} ${_mit_modname})
     list(APPEND _gss_root_hints "${_gss_PREFIX}")
     set(_gss_version "${_gss_VERSION}")
@@ -259,7 +259,7 @@ mark_as_advanced(
 
 if(GSS_FOUND)
   if(NOT TARGET CURL::gss)
-    add_library(CURL::gss INTERFACE IMPORTED)
+    add_library(CURL::gss INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::gss PROPERTIES
       INTERFACE_CURL_GSS_FLAVOUR "${_gss_flavour}"
       INTERFACE_LIBCURL_PC_MODULES "${_gss_pc_requires}"
diff --git a/CMake/FindGnuTLS.cmake b/CMake/FindGnuTLS.cmake
index 7f5b227dc2..39c8efd430 100644
--- a/CMake/FindGnuTLS.cmake
+++ b/CMake/FindGnuTLS.cmake
@@ -39,7 +39,7 @@ set(_gnutls_pc_requires "gnutls")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED GNUTLS_INCLUDE_DIR AND
    NOT DEFINED GNUTLS_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_gnutls ${_gnutls_pc_requires})
 endif()
 
@@ -81,7 +81,7 @@ endif()
 
 if(GNUTLS_FOUND)
   if(NOT TARGET CURL::gnutls)
-    add_library(CURL::gnutls INTERFACE IMPORTED)
+    add_library(CURL::gnutls INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::gnutls PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_gnutls_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_gnutls_CFLAGS}"
diff --git a/CMake/FindLDAP.cmake b/CMake/FindLDAP.cmake
index 8902b23f8d..642b8332a7 100644
--- a/CMake/FindLDAP.cmake
+++ b/CMake/FindLDAP.cmake
@@ -41,7 +41,7 @@ if(CURL_USE_PKGCONFIG AND
    NOT DEFINED LDAP_INCLUDE_DIR AND
    NOT DEFINED LDAP_LIBRARY AND
    NOT DEFINED LDAP_LBER_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_ldap ${_ldap_pc_requires})
 endif()
 
@@ -104,7 +104,7 @@ endif()
 
 if(LDAP_FOUND)
   if(NOT TARGET CURL::ldap)
-    add_library(CURL::ldap INTERFACE IMPORTED)
+    add_library(CURL::ldap INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::ldap PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_ldap_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_ldap_CFLAGS}"
diff --git a/CMake/FindLibbacktrace.cmake b/CMake/FindLibbacktrace.cmake
index 59da7fdf19..eb8b380ec8 100644
--- a/CMake/FindLibbacktrace.cmake
+++ b/CMake/FindLibbacktrace.cmake
@@ -48,7 +48,7 @@ if(LIBBACKTRACE_FOUND)
   set(_libbacktrace_LIBRARIES    ${LIBBACKTRACE_LIBRARY})
 
   if(NOT TARGET CURL::libbacktrace)
-    add_library(CURL::libbacktrace INTERFACE IMPORTED)
+    add_library(CURL::libbacktrace INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libbacktrace PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libbacktrace_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libbacktrace_CFLAGS}"
diff --git a/CMake/FindLibgsasl.cmake b/CMake/FindLibgsasl.cmake
index 7fbaa7128f..47428478b0 100644
--- a/CMake/FindLibgsasl.cmake
+++ b/CMake/FindLibgsasl.cmake
@@ -39,7 +39,7 @@ set(_libgsasl_pc_requires "libgsasl")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED LIBGSASL_INCLUDE_DIR AND
    NOT DEFINED LIBGSASL_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_libgsasl ${_libgsasl_pc_requires})
 endif()
 
@@ -80,7 +80,7 @@ endif()
 
 if(LIBGSASL_FOUND)
   if(NOT TARGET CURL::libgsasl)
-    add_library(CURL::libgsasl INTERFACE IMPORTED)
+    add_library(CURL::libgsasl INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libgsasl PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libgsasl_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libgsasl_CFLAGS}"
diff --git a/CMake/FindLibidn2.cmake b/CMake/FindLibidn2.cmake
index dc7873489a..ea214c3311 100644
--- a/CMake/FindLibidn2.cmake
+++ b/CMake/FindLibidn2.cmake
@@ -39,7 +39,7 @@ set(_libidn2_pc_requires "libidn2")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED LIBIDN2_INCLUDE_DIR AND
    NOT DEFINED LIBIDN2_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_libidn2 ${_libidn2_pc_requires})
 endif()
 
@@ -81,7 +81,7 @@ endif()
 
 if(LIBIDN2_FOUND)
   if(NOT TARGET CURL::libidn2)
-    add_library(CURL::libidn2 INTERFACE IMPORTED)
+    add_library(CURL::libidn2 INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libidn2 PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libidn2_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libidn2_CFLAGS}"
diff --git a/CMake/FindLibpsl.cmake b/CMake/FindLibpsl.cmake
index 2287e676d3..fed1c32264 100644
--- a/CMake/FindLibpsl.cmake
+++ b/CMake/FindLibpsl.cmake
@@ -39,7 +39,7 @@ set(_libpsl_pc_requires "libpsl")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED LIBPSL_INCLUDE_DIR AND
    NOT DEFINED LIBPSL_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_libpsl ${_libpsl_pc_requires})
 endif()
 
@@ -81,7 +81,7 @@ endif()
 
 if(LIBPSL_FOUND)
   if(NOT TARGET CURL::libpsl)
-    add_library(CURL::libpsl INTERFACE IMPORTED)
+    add_library(CURL::libpsl INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libpsl PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libpsl_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libpsl_CFLAGS}"
diff --git a/CMake/FindLibssh.cmake b/CMake/FindLibssh.cmake
index 3837d4de62..102829bcfa 100644
--- a/CMake/FindLibssh.cmake
+++ b/CMake/FindLibssh.cmake
@@ -40,7 +40,7 @@ set(_libssh_pc_requires "libssh")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED LIBSSH_INCLUDE_DIR AND
    NOT DEFINED LIBSSH_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_libssh ${_libssh_pc_requires})
 endif()
 
@@ -107,7 +107,7 @@ if(LIBSSH_FOUND)
   endif()
 
   if(NOT TARGET CURL::libssh)
-    add_library(CURL::libssh INTERFACE IMPORTED)
+    add_library(CURL::libssh INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libssh PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libssh_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libssh_CFLAGS}"
diff --git a/CMake/FindLibssh2.cmake b/CMake/FindLibssh2.cmake
index 22481d67ae..169cb3e294 100644
--- a/CMake/FindLibssh2.cmake
+++ b/CMake/FindLibssh2.cmake
@@ -40,11 +40,11 @@ set(_libssh2_pc_requires "libssh2")
 if(NOT DEFINED LIBSSH2_INCLUDE_DIR AND
    NOT DEFINED LIBSSH2_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_libssh2 ${_libssh2_pc_requires})
   endif()
   if(NOT _libssh2_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(libssh2 CONFIG QUIET)
+    find_package(libssh2 CONFIG QUIET GLOBAL)
   endif()
 endif()
 
@@ -106,7 +106,7 @@ endif()
 
 if(LIBSSH2_FOUND)
   if(NOT TARGET CURL::libssh2)
-    add_library(CURL::libssh2 INTERFACE IMPORTED)
+    add_library(CURL::libssh2 INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libssh2 PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libssh2_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libssh2_CFLAGS}"
diff --git a/CMake/FindLibuv.cmake b/CMake/FindLibuv.cmake
index 47d043cff6..62f620acd8 100644
--- a/CMake/FindLibuv.cmake
+++ b/CMake/FindLibuv.cmake
@@ -39,7 +39,7 @@ set(_libuv_pc_requires "libuv")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED LIBUV_INCLUDE_DIR AND
    NOT DEFINED LIBUV_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_libuv ${_libuv_pc_requires})
 endif()
 
@@ -91,7 +91,7 @@ endif()
 
 if(LIBUV_FOUND)
   if(NOT TARGET CURL::libuv)
-    add_library(CURL::libuv INTERFACE IMPORTED)
+    add_library(CURL::libuv INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::libuv PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_libuv_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_libuv_CFLAGS}"
diff --git a/CMake/FindMbedTLS.cmake b/CMake/FindMbedTLS.cmake
index 21a5f4aec7..6a781d6779 100644
--- a/CMake/FindMbedTLS.cmake
+++ b/CMake/FindMbedTLS.cmake
@@ -50,11 +50,11 @@ if(NOT DEFINED MBEDTLS_INCLUDE_DIR AND
    NOT DEFINED MBEDX509_LIBRARY AND
    NOT DEFINED MBEDCRYPTO_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_mbedtls ${_mbedtls_pc_requires})
   endif()
   if(NOT _mbedtls_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(MbedTLS CONFIG QUIET)
+    find_package(MbedTLS CONFIG QUIET GLOBAL)
   endif()
 endif()
 
@@ -126,7 +126,7 @@ endif()
 
 if(MBEDTLS_FOUND)
   if(NOT TARGET CURL::mbedtls)
-    add_library(CURL::mbedtls INTERFACE IMPORTED)
+    add_library(CURL::mbedtls INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::mbedtls PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_mbedtls_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_mbedtls_CFLAGS}"
diff --git a/CMake/FindNGHTTP2.cmake b/CMake/FindNGHTTP2.cmake
index f93113f404..52758cbc92 100644
--- a/CMake/FindNGHTTP2.cmake
+++ b/CMake/FindNGHTTP2.cmake
@@ -40,11 +40,11 @@ set(_nghttp2_pc_requires "libnghttp2")
 if(NOT DEFINED NGHTTP2_INCLUDE_DIR AND
    NOT DEFINED NGHTTP2_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_nghttp2 ${_nghttp2_pc_requires})
   endif()
   if(NOT _nghttp2_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(nghttp2 CONFIG QUIET)
+    find_package(nghttp2 CONFIG QUIET GLOBAL)
   endif()
 endif()
 
@@ -105,7 +105,7 @@ endif()
 
 if(NGHTTP2_FOUND)
   if(NOT TARGET CURL::nghttp2)
-    add_library(CURL::nghttp2 INTERFACE IMPORTED)
+    add_library(CURL::nghttp2 INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::nghttp2 PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_nghttp2_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_nghttp2_CFLAGS}"
diff --git a/CMake/FindNGHTTP3.cmake b/CMake/FindNGHTTP3.cmake
index 427c139f21..0467da267a 100644
--- a/CMake/FindNGHTTP3.cmake
+++ b/CMake/FindNGHTTP3.cmake
@@ -40,11 +40,11 @@ set(_nghttp3_pc_requires "libnghttp3")
 if(NOT DEFINED NGHTTP3_INCLUDE_DIR AND
    NOT DEFINED NGHTTP3_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_nghttp3 ${_nghttp3_pc_requires})
   endif()
   if(NOT _nghttp3_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(nghttp3 CONFIG QUIET)
+    find_package(nghttp3 CONFIG QUIET GLOBAL)
   endif()
 endif()
 
@@ -105,7 +105,7 @@ endif()
 
 if(NGHTTP3_FOUND)
   if(NOT TARGET CURL::nghttp3)
-    add_library(CURL::nghttp3 INTERFACE IMPORTED)
+    add_library(CURL::nghttp3 INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::nghttp3 PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_nghttp3_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_nghttp3_CFLAGS}"
diff --git a/CMake/FindNGTCP2.cmake b/CMake/FindNGTCP2.cmake
index e492916377..acf61fc7f1 100644
--- a/CMake/FindNGTCP2.cmake
+++ b/CMake/FindNGTCP2.cmake
@@ -77,12 +77,12 @@ set(_tried_pkgconfig FALSE)
 if(NOT DEFINED NGTCP2_INCLUDE_DIR AND
    NOT DEFINED NGTCP2_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_ngtcp2 ${_ngtcp2_pc_requires})
     set(_tried_pkgconfig TRUE)
   endif()
   if(NOT _ngtcp2_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(ngtcp2 CONFIG QUIET)
+    find_package(ngtcp2 CONFIG QUIET GLOBAL)
     # Skip using it if the crypto library target is not available
     if(ngtcp2_CONFIG AND
        NOT TARGET ngtcp2::${_crypto_library_lower}_static AND
@@ -175,7 +175,7 @@ endif()
 
 if(NGTCP2_FOUND)
   if(NOT TARGET CURL::ngtcp2)
-    add_library(CURL::ngtcp2 INTERFACE IMPORTED)
+    add_library(CURL::ngtcp2 INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::ngtcp2 PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_ngtcp2_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_ngtcp2_CFLAGS}"
diff --git a/CMake/FindNettle.cmake b/CMake/FindNettle.cmake
index c963180cba..f644f6b35d 100644
--- a/CMake/FindNettle.cmake
+++ b/CMake/FindNettle.cmake
@@ -39,7 +39,7 @@ set(_nettle_pc_requires "nettle")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED NETTLE_INCLUDE_DIR AND
    NOT DEFINED NETTLE_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_nettle ${_nettle_pc_requires})
 endif()
 
@@ -86,7 +86,7 @@ endif()
 
 if(NETTLE_FOUND)
   if(NOT TARGET CURL::nettle)
-    add_library(CURL::nettle INTERFACE IMPORTED)
+    add_library(CURL::nettle INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::nettle PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_nettle_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_nettle_CFLAGS}"
diff --git a/CMake/FindQuiche.cmake b/CMake/FindQuiche.cmake
index 137679b5b3..25204748a0 100644
--- a/CMake/FindQuiche.cmake
+++ b/CMake/FindQuiche.cmake
@@ -39,7 +39,7 @@ set(_quiche_pc_requires "quiche")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED QUICHE_INCLUDE_DIR AND
    NOT DEFINED QUICHE_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_quiche ${_quiche_pc_requires})
 endif()
 
@@ -69,7 +69,7 @@ endif()
 
 if(QUICHE_FOUND)
   if(NOT TARGET CURL::quiche)
-    add_library(CURL::quiche INTERFACE IMPORTED)
+    add_library(CURL::quiche INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::quiche PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_quiche_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_quiche_CFLAGS}"
diff --git a/CMake/FindRustls.cmake b/CMake/FindRustls.cmake
index 1e07565189..4dfcb982a8 100644
--- a/CMake/FindRustls.cmake
+++ b/CMake/FindRustls.cmake
@@ -39,7 +39,7 @@ set(_rustls_pc_requires "rustls")
 if(CURL_USE_PKGCONFIG AND
    NOT DEFINED RUSTLS_INCLUDE_DIR AND
    NOT DEFINED RUSTLS_LIBRARY)
-  find_package(PkgConfig QUIET)
+  find_package(PkgConfig QUIET GLOBAL)
   pkg_check_modules(_rustls ${_rustls_pc_requires})
 endif()
 
@@ -105,7 +105,7 @@ if(RUSTLS_FOUND)
   endif()
 
   if(NOT TARGET CURL::rustls)
-    add_library(CURL::rustls INTERFACE IMPORTED)
+    add_library(CURL::rustls INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::rustls PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_rustls_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_rustls_CFLAGS}"
diff --git a/CMake/FindWolfSSL.cmake b/CMake/FindWolfSSL.cmake
index 964320ad6b..11dab35dad 100644
--- a/CMake/FindWolfSSL.cmake
+++ b/CMake/FindWolfSSL.cmake
@@ -48,11 +48,11 @@ set(_wolfssl_pc_requires "wolfssl")
 if(NOT DEFINED WOLFSSL_INCLUDE_DIR AND
    NOT DEFINED WOLFSSL_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_wolfssl ${_wolfssl_pc_requires})
   endif()
   if(NOT _wolfssl_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(wolfssl CONFIG QUIET)
+    find_package(wolfssl CONFIG QUIET GLOBAL)
   endif()
 endif()
 
@@ -124,7 +124,7 @@ if(WOLFSSL_FOUND)
   endif()
 
   if(NOT TARGET CURL::wolfssl)
-    add_library(CURL::wolfssl INTERFACE IMPORTED)
+    add_library(CURL::wolfssl INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::wolfssl PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_wolfssl_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_wolfssl_CFLAGS}"
diff --git a/CMake/FindZstd.cmake b/CMake/FindZstd.cmake
index 8dc620a100..c572d3817d 100644
--- a/CMake/FindZstd.cmake
+++ b/CMake/FindZstd.cmake
@@ -49,11 +49,11 @@ set(_zstd_pc_requires "libzstd")
 if(NOT DEFINED ZSTD_INCLUDE_DIR AND
    NOT DEFINED ZSTD_LIBRARY)
   if(CURL_USE_PKGCONFIG)
-    find_package(PkgConfig QUIET)
+    find_package(PkgConfig QUIET GLOBAL)
     pkg_check_modules(_zstd ${_zstd_pc_requires})
   endif()
   if(NOT _zstd_FOUND AND CURL_USE_CMAKECONFIG)
-    find_package(Zstd CONFIG QUIET)
+    find_package(Zstd CONFIG QUIET GLOBAL)
     # Skip using if older than v1.4.5
     if(Zstd_CONFIG AND
        NOT TARGET zstd::libzstd_static AND
@@ -132,7 +132,7 @@ endif()
 
 if(ZSTD_FOUND)
   if(NOT TARGET CURL::zstd)
-    add_library(CURL::zstd INTERFACE IMPORTED)
+    add_library(CURL::zstd INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::zstd PROPERTIES
       INTERFACE_LIBCURL_PC_MODULES "${_zstd_pc_requires}"
       INTERFACE_COMPILE_OPTIONS "${_zstd_CFLAGS}"
diff --git a/CMake/Macros.cmake b/CMake/Macros.cmake
index 5e26c38469..408fa3d102 100644
--- a/CMake/Macros.cmake
+++ b/CMake/Macros.cmake
@@ -64,15 +64,15 @@ macro(curl_dependency_option _option_name _find_name _desc_name)
 
   if(${_option_name} STREQUAL "AUTO")
     if(_find_name STREQUAL "ZLIB")
-      find_package(${_find_name})
+      find_package(${_find_name} GLOBAL)
     else()
-      find_package(${_find_name} MODULE)
+      find_package(${_find_name} MODULE GLOBAL)
     endif()
   elseif(${_option_name})
     if(_find_name STREQUAL "ZLIB")
-      find_package(${_find_name} REQUIRED)
+      find_package(${_find_name} REQUIRED GLOBAL)
     else()
-      find_package(${_find_name} MODULE REQUIRED)
+      find_package(${_find_name} MODULE REQUIRED GLOBAL)
     endif()
   else()
     string(TOUPPER "${_find_name}" _find_name_upper)
diff --git a/CMake/curl-config.in.cmake b/CMake/curl-config.in.cmake
index 317477197f..9e06b8f889 100644
--- a/CMake/curl-config.in.cmake
+++ b/CMake/curl-config.in.cmake
@@ -47,7 +47,7 @@ if("@USE_OPENSSL@")
   endif()
   # Define lib duplicate to fixup lib order for GCC binutils ld in static builds
   if(TARGET OpenSSL::Crypto AND NOT TARGET CURL::OpenSSL_Crypto)
-    add_library(CURL::OpenSSL_Crypto INTERFACE IMPORTED)
+    add_library(CURL::OpenSSL_Crypto INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::OpenSSL_Crypto PROPERTIES INTERFACE_LINK_LIBRARIES OpenSSL::Crypto)
   endif()
 endif()
@@ -55,7 +55,7 @@ if("@HAVE_LIBZ@")
   find_dependency(ZLIB "@ZLIB_VERSION_MAJOR@")
   # Define lib duplicate to fixup lib order for GCC binutils ld in static builds
   if(TARGET ZLIB::ZLIB AND NOT TARGET CURL::ZLIB)
-    add_library(CURL::ZLIB INTERFACE IMPORTED)
+    add_library(CURL::ZLIB INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::ZLIB PROPERTIES INTERFACE_LINK_LIBRARIES ZLIB::ZLIB)
   endif()
 endif()
@@ -152,7 +152,7 @@ set(CMAKE_MODULE_PATH ${_curl_cmake_module_path_save})
 
 # Define lib duplicate to fixup lib order for GCC binutils ld in static builds
 if(WIN32 AND NOT TARGET CURL::win32_winsock)
-  add_library(CURL::win32_winsock INTERFACE IMPORTED)
+  add_library(CURL::win32_winsock INTERFACE IMPORTED GLOBAL)
   set_target_properties(CURL::win32_winsock PROPERTIES INTERFACE_LINK_LIBRARIES "ws2_32")
 endif()
 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5dd9c7aa70..0259001665 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -409,7 +409,7 @@ set(CURL_LIBS "")
 
 if(ENABLE_ARES)
   set(USE_ARES 1)
-  find_package(Cares MODULE REQUIRED)
+  find_package(Cares MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::cares)
 endif()
 
@@ -558,7 +558,7 @@ endif()
 
 option(CURL_LINT "Run lint checks while building" OFF)
 
-find_package(Perl)
+find_package(Perl GLOBAL)
 
 if(PERL_EXECUTABLE)
   add_custom_target(curl-ca-bundle
@@ -644,7 +644,7 @@ if(DOS OR AMIGA)
 endif()
 
 if(NOT WIN32)
-  find_package(Threads)
+  find_package(Threads GLOBAL)
   if(CMAKE_USE_PTHREADS_INIT)
     set(HAVE_THREADS_POSIX 1)
     list(APPEND CURL_NETWORK_AND_TIME_LIBS Threads::Threads)
@@ -820,7 +820,7 @@ if(_use_core_foundation_and_core_services)
 endif()
 
 if(CURL_USE_OPENSSL)
-  find_package(OpenSSL REQUIRED)
+  find_package(OpenSSL REQUIRED GLOBAL)
   set(_ssl_enabled ON)
   set(USE_OPENSSL ON)
 
@@ -848,7 +848,7 @@ if(CURL_USE_OPENSSL)
 
   if(HAVE_BORINGSSL OR HAVE_AWSLC)
     if(NOT MSVC AND NOT ANDROID)  # BoringSSL/AWS-LC MSVC builds use native Windows threads
-      find_package(Threads)
+      find_package(Threads GLOBAL)
       if(CMAKE_USE_PTHREADS_INIT)
         set(HAVE_THREADS_POSIX_BORINGSSL 1)
         list(APPEND CURL_NETWORK_AND_TIME_LIBS Threads::Threads)
@@ -880,7 +880,7 @@ if(CURL_USE_OPENSSL)
 endif()
 
 if(CURL_USE_MBEDTLS)
-  find_package(MbedTLS MODULE REQUIRED)
+  find_package(MbedTLS MODULE REQUIRED GLOBAL)
   if(MBEDTLS_VERSION VERSION_LESS 3.2.0)
     message(FATAL_ERROR "mbedTLS v3.2.0 or newer is required.")
   endif()
@@ -905,7 +905,7 @@ if(CURL_USE_MBEDTLS)
 endif()
 
 if(CURL_USE_WOLFSSL)
-  find_package(WolfSSL MODULE REQUIRED)
+  find_package(WolfSSL MODULE REQUIRED GLOBAL)
   set(_ssl_enabled ON)
   set(USE_WOLFSSL ON)
   list(APPEND CURL_LIBS CURL::wolfssl)
@@ -923,9 +923,9 @@ if(CURL_USE_WOLFSSL)
 endif()
 
 if(CURL_USE_GNUTLS)
-  find_package(GnuTLS MODULE REQUIRED)
+  find_package(GnuTLS MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::gnutls)
-  find_package(Nettle MODULE REQUIRED)
+  find_package(Nettle MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::nettle)
   set(_ssl_enabled ON)
   set(USE_GNUTLS ON)
@@ -946,7 +946,7 @@ if(CURL_USE_GNUTLS)
 endif()
 
 if(CURL_USE_RUSTLS)
-  find_package(Rustls MODULE REQUIRED)
+  find_package(Rustls MODULE REQUIRED GLOBAL)
   set(_ssl_enabled ON)
   set(USE_RUSTLS ON)
   list(APPEND CURL_LIBS CURL::rustls)
@@ -1120,7 +1120,7 @@ endif()
 
 option(USE_NGHTTP2 "Use nghttp2 library" ON)
 if(USE_NGHTTP2)
-  find_package(NGHTTP2 MODULE)
+  find_package(NGHTTP2 MODULE GLOBAL)
   if(NGHTTP2_FOUND)
     list(APPEND CURL_LIBS CURL::nghttp2)
   else()
@@ -1134,33 +1134,33 @@ if(USE_NGTCP2)
     message(FATAL_ERROR "MultiSSL cannot be enabled with HTTP/3 and vice versa.")
   elseif(USE_OPENSSL OR USE_WOLFSSL)
     if(USE_WOLFSSL)
-      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "wolfSSL")
+      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "wolfSSL" GLOBAL)
     elseif(HAVE_BORINGSSL OR HAVE_AWSLC)
-      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "BoringSSL")
+      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "BoringSSL" GLOBAL)
     elseif(OPENSSL_VERSION VERSION_GREATER_EQUAL 3.5.0)
-      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "ossl")
+      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "ossl" GLOBAL)
       if(NGTCP2_VERSION VERSION_LESS 1.12.0)
         message(FATAL_ERROR "ngtcp2 1.12.0 or upper required for OpenSSL")
       endif()
       set(OPENSSL_QUIC_API2 1)
     elseif(HAVE_LIBRESSL)
-      find_package(NGTCP2 MODULE COMPONENTS "LibreSSL")
+      find_package(NGTCP2 MODULE COMPONENTS "LibreSSL" GLOBAL)
       if(NOT NGTCP2_FOUND)
-        find_package(NGTCP2 MODULE REQUIRED COMPONENTS "quictls")  # for ngtcp2 <1.15.0
+        find_package(NGTCP2 MODULE REQUIRED COMPONENTS "quictls" GLOBAL)  # for ngtcp2 <1.15.0
       endif()
     else()
-      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "quictls")
+      find_package(NGTCP2 MODULE REQUIRED COMPONENTS "quictls" GLOBAL)
       set(_openssl "quictls")
     endif()
     curl_openssl_check_quic()
   elseif(USE_GNUTLS)
-    find_package(NGTCP2 MODULE REQUIRED "GnuTLS")
+    find_package(NGTCP2 MODULE REQUIRED "GnuTLS" GLOBAL)
   else()
     message(FATAL_ERROR "ngtcp2 requires a supported TLS-backend")
   endif()
   list(APPEND CURL_LIBS CURL::ngtcp2)
 
-  find_package(NGHTTP3 MODULE REQUIRED)
+  find_package(NGHTTP3 MODULE REQUIRED GLOBAL)
   set(USE_NGHTTP3 ON)
   list(APPEND CURL_LIBS CURL::nghttp3)
 endif()
@@ -1172,7 +1172,7 @@ if(USE_QUICHE)
   elseif(CURL_WITH_MULTI_SSL)
     message(FATAL_ERROR "MultiSSL cannot be enabled with HTTP/3 and vice versa.")
   endif()
-  find_package(Quiche MODULE REQUIRED)
+  find_package(Quiche MODULE REQUIRED GLOBAL)
   if(NOT HAVE_BORINGSSL)
     message(FATAL_ERROR "quiche requires BoringSSL")
   endif()
@@ -1208,7 +1208,7 @@ if(NOT CURL_DISABLE_LDAP)
     if(USE_OPENSSL)
       list(APPEND CMAKE_REQUIRED_LIBRARIES OpenSSL::SSL OpenSSL::Crypto)
     endif()
-    find_package(LDAP MODULE)
+    find_package(LDAP MODULE GLOBAL)
     if(LDAP_FOUND)
       set(HAVE_LBER_H 1)
       list(PREPEND CURL_LIBS CURL::ldap)
@@ -1276,7 +1276,7 @@ option(USE_LIBIDN2 "Use libidn2 for IDN support" ON)
 set(HAVE_IDN2_H OFF)
 set(HAVE_LIBIDN2 OFF)
 if(USE_LIBIDN2 AND NOT USE_APPLE_IDN AND NOT USE_WIN32_IDN)
-  find_package(Libidn2 MODULE)
+  find_package(Libidn2 MODULE GLOBAL)
   if(LIBIDN2_FOUND)
     list(PREPEND CURL_LIBS CURL::libidn2)
     set(HAVE_IDN2_H 1)
@@ -1289,7 +1289,7 @@ option(CURL_USE_LIBPSL "Use libpsl" ON)
 mark_as_advanced(CURL_USE_LIBPSL)
 set(USE_LIBPSL OFF)
 if(CURL_USE_LIBPSL)
-  find_package(Libpsl MODULE REQUIRED)
+  find_package(Libpsl MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::libpsl)
   set(USE_LIBPSL ON)
 endif()
@@ -1299,7 +1299,7 @@ option(CURL_USE_LIBSSH2 "Use libssh2" ON)
 mark_as_advanced(CURL_USE_LIBSSH2)
 set(USE_LIBSSH2 OFF)
 if(CURL_USE_LIBSSH2)
-  find_package(Libssh2 MODULE)
+  find_package(Libssh2 MODULE GLOBAL)
   if(LIBSSH2_FOUND)
     list(PREPEND CURL_LIBS CURL::libssh2)  # keep it before TLS-crypto, compression
     set(USE_LIBSSH2 ON)
@@ -1310,7 +1310,7 @@ endif()
 option(CURL_USE_LIBSSH "Use libssh" OFF)
 mark_as_advanced(CURL_USE_LIBSSH)
 if(NOT USE_LIBSSH2 AND CURL_USE_LIBSSH)
-  find_package(Libssh MODULE REQUIRED)
+  find_package(Libssh MODULE REQUIRED GLOBAL)
   list(PREPEND CURL_LIBS CURL::libssh)  # keep it before TLS-crypto, compression
   set(USE_LIBSSH ON)
 endif()
@@ -1318,7 +1318,7 @@ endif()
 option(CURL_USE_GSASL "Use libgsasl" OFF)
 mark_as_advanced(CURL_USE_GSASL)
 if(CURL_USE_GSASL)
-  find_package(Libgsasl MODULE REQUIRED)
+  find_package(Libgsasl MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::libgsasl)
   set(USE_GSASL ON)
 endif()
@@ -1327,7 +1327,7 @@ option(CURL_USE_GSSAPI "Use GSSAPI implementation" OFF)
 mark_as_advanced(CURL_USE_GSSAPI)
 
 if(CURL_USE_GSSAPI)
-  find_package(GSS MODULE)
+  find_package(GSS MODULE GLOBAL)
 
   set(HAVE_GSSAPI ${GSS_FOUND})
   if(GSS_FOUND)
@@ -1353,7 +1353,7 @@ if(CURL_USE_LIBBACKTRACE)
   if(NOT CMAKE_BUILD_TYPE MATCHES "(Debug|RelWithDebInfo)")
     message(FATAL_ERROR "libbacktrace requires debug information")
   endif()
-  find_package(Libbacktrace MODULE REQUIRED)
+  find_package(Libbacktrace MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::libbacktrace)
   set(USE_BACKTRACE ON)
 endif()
@@ -1364,7 +1364,7 @@ if(CURL_USE_LIBUV)
   if(NOT ENABLE_DEBUG)
     message(FATAL_ERROR "Using libuv without debug support enabled is useless")
   endif()
-  find_package(Libuv MODULE REQUIRED)
+  find_package(Libuv MODULE REQUIRED GLOBAL)
   list(APPEND CURL_LIBS CURL::libuv)
   set(USE_LIBUV ON)
   set(HAVE_UV_H ON)
@@ -1864,17 +1864,17 @@ endif()
 # to consume libcurl, regardless of the compiler used to build libcurl itself.
 if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
   if(USE_OPENSSL AND TARGET OpenSSL::Crypto)
-    add_library(CURL::OpenSSL_Crypto INTERFACE IMPORTED)
+    add_library(CURL::OpenSSL_Crypto INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::OpenSSL_Crypto PROPERTIES INTERFACE_LINK_LIBRARIES OpenSSL::Crypto)
     list(APPEND CURL_LIBS CURL::OpenSSL_Crypto)
   endif()
   if(HAVE_LIBZ AND TARGET ZLIB::ZLIB)
-    add_library(CURL::ZLIB INTERFACE IMPORTED)
+    add_library(CURL::ZLIB INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::ZLIB PROPERTIES INTERFACE_LINK_LIBRARIES ZLIB::ZLIB)
     list(APPEND CURL_LIBS CURL::ZLIB)
   endif()
   if(WIN32)
-    add_library(CURL::win32_winsock INTERFACE IMPORTED)
+    add_library(CURL::win32_winsock INTERFACE IMPORTED GLOBAL)
     set_target_properties(CURL::win32_winsock PROPERTIES INTERFACE_LINK_LIBRARIES "ws2_32")
     list(APPEND CURL_LIBS CURL::win32_winsock)
   endif()
